test/iterator.js

```diff
@@ -241,6 +241,12 @@ metatests.test('Iterator.take', test => {
   test.end();
 });
 
+metatests.testSync('Iterator.takeWhile', test => {
+  const it = iter(array).takeWhile(x => x < 3);
+  test.strictSame(it.toArray(), [1, 2]);
+  test.assert(it.next().done);
```

Comment:
What is the point of this assert?
If `done` value is `false` here, then `toArray`, in the line before that, isn't working correctly, so you are testing the `toArray` method with this assert.

Reply:
I test that after we exited once iterator won't continue again.
This can be easily simulated with returning `done: false` in the loop but not setting `this.done` variable and having array of sort [1, 2, 3, 1, 1] for this case.

Reply:
Okay, I'm fine with it, but that still doesn't guarantee that if you call `next()` method one more time it won't continue again, meaning that to test it fully we need to add an infinite amount of checks for the `next()` result, which is impossible to do.

Reply:
Yeah, but at least we check that we didn't make an unintentional error of missing to check/set `done`. Obviously, this won't protect against a manually crafted implementation that will avoid this check.